src/bindings/js/node/src/core_wrap.cpp

```diff
@@ -51,46 +56,216 @@ Napi::Value CoreWrap::read_model_async(const Napi::CallbackInfo& info) {
     std::string bin_path;
 
     try {
-        switch(info.Length()) {
-            case 2:
-                if (!info[1].IsString())
-                    throw std::runtime_error("Second argument should be string");
-
-                bin_path = info[1].ToString();
-                [[fallthrough]];
-            case 1:
-                if (!info[0].IsString())
-                    throw std::runtime_error("First argument should be string");
-
-                model_path = info[0].ToString();
-                break;
-
-            default:
-                throw std::runtime_error("Invalid number of arguments -> " + std::to_string(info.Length()));
+        switch (info.Length()) {
+        case 2:
+            if (!info[1].IsString())
+                throw std::runtime_error("Second argument should be string");
+
+            bin_path = info[1].ToString();
+            [[fallthrough]];
+        case 1:
+            if (!info[0].IsString())
+                throw std::runtime_error("First argument should be string");
+
+            model_path = info[0].ToString();
+            break;
+
+        default:
+            throw std::runtime_error("Invalid number of arguments -> " + std::to_string(info.Length()));
         }
 
         ReaderWorker* _readerWorker = new ReaderWorker(info.Env(), model_path, bin_path);
         _readerWorker->Queue();
 
         return _readerWorker->GetPromise();
-    } catch (std::runtime_error &err) {
+    } catch (std::runtime_error& err) {
         reportError(info.Env(), err.what());
 
         return Napi::Value();
     }
 }
 
-Napi::Value CoreWrap::compile_model(const Napi::CallbackInfo& info) {
-    if (info.Length() != 2) {
+Napi::Value CoreWrap::compile_model_sync_helper(const Napi::CallbackInfo& info,
```

Comment:
1. Method has unexplainable name
2. These overloads shouldn't be available outside (protected modifier can resolve this issue)

Reply:
Similarly to infer() method I changed those methods to compile_model_sync_dispatch() and compile_model_sync()